hare4/hare_test.go

```diff
@@ -502,7 +499,11 @@ func (cl *lockstepCluster) genProposals(lid types.LayerID, skipNodes ...int) {
 		proposal.SetID(id)
 		proposal.Ballot.SetID(bid)
 		var vrf types.VrfSignature
-		cl.t.rng.Read(vrf[:])
+		if !cl.collidingProposals {
+			// if we want non-colliding proposals we copy from the rng
+			// otherwise it is kept as an array of zeroes
+			cl.t.rng.Read(vrf[:])
+		}
```

Comment:
How about just overwriting the first bytes?
```suggestion
		cl.t.rng.Read(vrf[:])
		if cl.collidingProposals {
			copy(vrf[:4], []byte("1234"))
		}
```